panic: runtime error: invalid memory address or nil pointer dereference

When running example krakend.json we get following error 

```
[GIN-debug] Listening and serving HTTP on XXXX
[GIN-debug] [ERROR] listen tcp: address XXXX: missing port in address
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x45d3279]

goroutine 1 [running]:
github.com/luraproject/lura/v2/router/gin.ginRouter.Run({{0xc00037d040, {0x0, 0x0, 0x0}, 0xc0003f39c0, {0x47e9440, 0xc000460840}, {0x47fea60, 0xc000427bf0}, 0x0}, ...}, ...)
	/Users/rajan/.gvm/pkgsets/go1.17/global/pkg/mod/github.com/luraproject/lura/v2@v2.0.0/router/gin/router.go:111 +0x219
main.main()
	/Users/rajan/Desktop/projects/krakend-martian/example/main.go:51 +0x6f8
exit status 2
```

I updated the `krakend.json` version to following 
```
"version": 3,
```

Go version: `go1.17`
OS: MacOS 12.1
When I run the go file using `go run example/main.go -c example/krakend.json -d -p 8080` I get above error.

**EDIT**:

I also tried to checkout `v1.4.0` and tried running the old version but it did not work either and failed with the same error.

--- a/example/krakend.json
+++ b/example/krakend.json
@@ -5,12 +5,12 @@
     "cache_ttl": 3600,
     "timeout": "3s",
     "extra_config": {
-      "github_com/devopsfaith/krakend-gologging": {
-        "level":  "DEBUG",
-        "prefix": "[KRAKEND]",
-        "syslog": false,
-        "stdout": true
-      }
+        "github_com/devopsfaith/krakend-gologging": {
+            "level": "DEBUG",
+            "prefix": "[KRAKEND]",
+            "syslog": false,
+            "stdout": true
+        }
     },
     "endpoints": [
         {
@@ -18,19 +18,26 @@
             "method": "GET",
             "backend": [
                 {
+                    "method": "GET",
                     "host": [
-                        "http://127.0.0.1:8000"
+                        "https://api.github.com"
                     ],
-                    "url_pattern": "/__debug/supu",
+                    "url_pattern": "/users/devopsfaith",
                     "extra_config": {
                         "github.com/devopsfaith/krakend-martian": {
                             "fifo.Group": {
-                                "scope": ["request", "response"],
+                                "scope": [
+                                    "request",
+                                    "response"
+                                ],
                                 "aggregateErrors": true,
                                 "modifiers": [
                                     {
                                         "body.Modifier": {
-                                            "scope": ["request","response"],
+                                            "scope": [
+                                                "request",
+                                                "response"
+                                            ],
                                             "body": "eyJtc2ciOiJ5b3Ugcm9jayEifQ=="
                                         }
                                     }
@@ -56,7 +63,7 @@
                                     "request",
                                     "response"
                                 ],
-                                "rootPath": "/etc/krakend/static",
+                                "rootPath": "example/static",
                                 "explicitPaths": {
                                     "/": "/health.json"
                                 }

--- a/example/krakend.json
+++ b/example/krakend.json
@@ -1,5 +1,5 @@
 {
-    "version": 2,
+    "version": 3,
     "name": "My lovely gateway",
     "port": 8080,
     "cache_ttl": 3600,

--- a/example/main.go
+++ b/example/main.go
@@ -4,6 +4,9 @@ import (
 	"context"
 	"flag"
 	"log"
+	"os"
+	"os/signal"
+	"syscall"
 
 	gologging "github.com/devopsfaith/krakend-gologging/v2"
 	martian "github.com/devopsfaith/krakend-martian/v2"
@@ -12,6 +15,7 @@ import (
 	"github.com/luraproject/lura/v2/proxy"
 	krakendgin "github.com/luraproject/lura/v2/router/gin"
 	"github.com/luraproject/lura/v2/transport/http/client"
+	"github.com/luraproject/lura/v2/transport/http/server"
 )
 
 func main() {
@@ -35,10 +39,6 @@ func main() {
 		log.Fatal("ERROR:", err.Error())
 	}
 
-	logger.Debug("config:", serviceConfig)
-
-	ctx, cancel := context.WithCancel(context.Background())
-
 	backendFactory := martian.NewBackendFactory(logger, client.DefaultHTTPRequestExecutor(client.NewHTTPClient))
 
 	routerFactory := krakendgin.NewFactory(krakendgin.Config{
@@ -46,9 +46,24 @@ func main() {
 		Logger:         logger,
 		HandlerFactory: krakendgin.EndpointHandler,
 		ProxyFactory:   proxy.NewDefaultFactory(backendFactory, logger),
+		RunServer:      server.RunServer,
 	})
+
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go func() {
+		select {
+		case sig := <-sigs:
+			logger.Info("Signal intercepted:", sig)
+			cancel()
+		case <-ctx.Done():
+		}
+	}()
 
 	routerFactory.NewWithContext(ctx).Run(serviceConfig)
 
-	cancel()
+	// cancel()
 }

